Fixing the 'device_instance' being None in some partition places

Torxed opened the pull request:
One minor issue was that `add_partition` was defined to return `-> None`.
Another issue might be uuid comparison using small/capital letters when comparing, so unified that.
Added a bunch of debugging and placed an exception in case `add_partition` doesn't return anything at all.

dylanmtaylor commented on archinstall/lib/disk/filesystem.py:
```diff
@@ -181,31 +189,41 @@ def add_partition(self, partition_type :str, start :str, end :str, partition_for
 				else:
 					count += 1
-					log(f"Could not get UUID for partition. Waiting for the {count} time",level=logging.DEBUG)
+					log(f"Could not get UUID for partition. Waiting for the {count} time",level=logging.WARNING, fg="yellow")
```
Can we make the text something like 'Retry attempt 1 of 10.' instead of Waiting for the {count} time"? It's more grammatically correct since we're not doing '1st', '2nd', etc.

Torxed commented:
Found the possible culprit: `filesystem.partprobe` some times couldn't execute properly for different reasons.
This was passed silently allowing for `add_partition` (who calls `partprobe`) to return `None` when that happened.

Added an exception to `add_partition` so that it can't return nothing, but also added more debugging in case something like this ever happens again.

Torxed commented:
We also had to change and make sure `Partition()` objects are properly configured with a `BlockDevice()` so that subsequent calls to `.size` and similar attributes doesn't break.
